example/extensions/lib_subgraph/subgraph_lib.cc

```diff
@@ -0,0 +1,253 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2019 by Contributors
+ * \file subgraph_lib.cc
+ * \brief subgraph operator implementation library file
+ */
+
+#include <iostream>
+#include <algorithm>
+#include "lib_api.h"
+
+MXReturnValue parseAttrs(std::map<std::string, std::string> attrs,
+                         int* num_in, int* num_out) {
+  *num_in = 1;
+  *num_out = 1;
+  if (attrs.count(SUBGRAPH_SYM_JSON)) {
+    // example of subgraph json parsing
+    JsonParser jp;
+    JsonVal val = jp.parse_to_json(attrs[SUBGRAPH_SYM_JSON]);
+    int input = 0;
+    for (auto &item : val.map[JsonVal("nodes")].list) {
+      if (item.map[JsonVal("op")].str == "null")
+        input++;
+    }
+    int output = val.map[JsonVal("heads")].list.size();
+    *num_in = input;
+    *num_out = output;
+  }
+  return MX_SUCCESS;
+}
+
+MXReturnValue inferType(std::map<std::string, std::string> attrs,
+                        std::vector<int> &intypes,
+                        std::vector<int> &outtypes) {
+  outtypes[0] = intypes[0];
+  return MX_SUCCESS;
+}
+
+MXReturnValue inferShape(std::map<std::string, std::string> attrs,
+                         std::vector<std::vector<unsigned int>> &inshapes,
+                         std::vector<std::vector<unsigned int>> &outshapes) {
+  outshapes[0] = inshapes[0];
+  return MX_SUCCESS;
+}
+
+/* function to execute log operator on floats */
+void myLog(MXTensor &in, MXTensor &out) {
+  float* inp = in.data<float>();
+  float* outp = out.data<float>();
+  for (int i = 0; i < in.size(); i++) {
```

Comment:
changed to int64_t since this is what the size function returns:
https://github.com/apache/incubator-mxnet/blob/f86a8d10074f983b990ef420b96d664e911ee1bf/include/mxnet/lib_api.h#L271-L278